### PR TITLE
Laboratory work #1, Lilia Sabirova - 23FPL1

### PR DESCRIPTION
**Pull request by SabirovaLilia**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -20,6 +20,15 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not isinstance(text, str):
+        return None
+    text = text.lower()
+    cleaned_text = ''
+    for i in text:
+        if i.isalpha():
+            cleaned_text += i
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:29]: concatenation of strings is a bad way, especially then you return list

File: seminars/practice_2_strings.py [hunk 2]
```diff
@@ -47,19 +47,19 @@
 # TASKS
 
 # Task 1:
-def multiply_string(input_string: str, how_many: int) -> str:
+def multiply_string(text: str, number: int) -> str:    # в скобках подсказка, что в функцию нажо передать строчку и число
```
Comment (artyomtugaryov) [inline seminars/practice_2_strings.py:50]: checkout this changes

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (SabirovaLilia) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -34,6 +43,24 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        return None
+    elif isinstance(tokens, list):
+        for element in tokens:
+            if not isinstance(element, str):
+                return None
+    length = len(tokens)
+    freq_dictionary = dict()
+
+    for letter in tokens:
+        if letter in freq_dictionary:
```
[the remark this answers is not in the document]
Comment (SabirovaLilia) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -34,6 +43,24 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+            freq_dictionary[letter] += 1
+        else:
```
[the remark this answers is not in the document]
Comment (SabirovaLilia) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -64,6 +97,23 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
+    elif isinstance(actual, list):
+        for element in actual:
+            if not isinstance(element, float) and not isinstance(element, int):
+                return None
+    if len(predicted) != len(actual):
+        return None
+    sum_mse = 0
+    for i in range(len(predicted)):
+        sum_mse += (predicted[i] - actual[i]) ** 2
+    result_mse = sum_mse / len(predicted)
```
[the remark this answers is not in the document]
Comment (SabirovaLilia) [reply lab_1_classify_by_unigrams/main.py:112]: done

File: seminars/practice_2_strings.py (hunk 2 above)
Comment (SabirovaLilia) [reply seminars/practice_2_strings.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -49,6 +73,12 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
 
     In case of corrupt input arguments, None is returned
     """
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    language_profile_dictionary = dict()
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:78]: You car return `as is` without creating a variable

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -34,6 +42,22 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        return None
+
+    length = len(tokens)
+    freq_dictionary = {}
+
+    for letter in tokens:
+        if not isinstance(letter, str):
+            return None
+        if letter not in freq_dictionary:
+            freq_dictionary[letter] = float(0)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:55]: 0.0

Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:55]: instead of `float`

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -84,6 +131,51 @@ def compare_profiles(
...
+    # если значение частоты в словаре отсутсвует, то 0
+    new_list_unknown = [
+        frequency_unknown_profile.get(element, 0) for element in sorted_tokens
+    ]
+
+    new_list_to_compare = [
+        frequency_profile_to_compare.get(element, 0) for element in sorted_tokens
+    ]
+
+    some_mse = calculate_mse(new_list_unknown, new_list_to_compare)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:177]: do not create the variable

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (SabirovaLilia) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (SabirovaLilia) [reply lab_1_classify_by_unigrams/main.py]: done

**Pull request closed without merging**